File: 1/2020.11.04/621.Task Scheduler/java/solution.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] map = new int[26];
        for (char task : tasks) {
            map[task - 'A']++;
        }
        Arrays.sort(map);
        int maxVal = map[25] - 1;
        int slots = maxVal * n;
        for (int i = 24; i >= 0 && map[i] > 0; i--) {
            slots -= Math.min(map[i], maxVal);
        }
        return slots > 0 ? slots + tasks.length : tasks.length;
    }
}

File: 1/2020.11.04/621.Task Scheduler/python/solution.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        arr = [0 for _ in range(26)]
        for task in tasks:
            arr[ord(task) - ord('A')] += 1
        arr.sort(reverse=True)
        max_val = arr[0] - 1
        slots = max_val * n
        for i in range(1, 26):
            if arr[i] == 0:
                break
            slots -= min(arr[i], max_val)
        return slots + len(tasks) if slots > 0 else len(tasks)
